### PR TITLE
HPCC-21441 Make sure audience and message class is sensible

### PR DESCRIPTION
**Pull request by shamser**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by shamser** [comment]
@ghalliday I have replaced ERRLOG with UERRLOG as it is not clear that that it is for USER audience messages. It seems that we have been using ERRLOG for just about every error message without thinking - now we'll need to think about which one to use (UERRLOG for user messages, OERRLOG for sysadmin messages, IERRLOG for internal/programmer messages)

File: common/remote/sockfile.cpp [hunk 1]
```diff
@@ -1398,7 +1398,7 @@ protected: friend class CRemoteFileIO;
             else
                 msg.append("ERROR #").append(errCode);
 #ifdef _DEBUG
-            ERRLOG("%s",msg.str());
+            OERRLOG("%s",msg.str());
```
Comment (ghalliday) [inline common/remote/sockfile.cpp]: I think better as ERRLOG - this may be things like file not found.  @jakesmith what do you think?

File: common/roxiecommlib/roxiecommunicationclient.cpp [hunk 2]
```diff
@@ -306,7 +306,7 @@ class CRoxieCommunicationClient: implements IRoxieCommunicationClient, public CI
             StringBuffer err;
             err.appendf("%d ", errCode);
             e->errorMessage(err);
-            DBGLOG("ERROR loading query info directly to roxie %s", err.str());
+            OERRLOG("ERROR loading query info directly to roxie %s", err.str());
```
Comment (ghalliday) [inline common/roxiecommlib/roxiecommunicationclient.cpp]: Probably ERRLOG (comment above indicates it is not fatal, so operator should not need to know)

File: common/thorhelper/roxiehelper.cpp [hunk 3]
```diff
@@ -1666,7 +1666,7 @@ void CSafeSocket::sendSoapException(IException *E, const char *queryName)
     {
         StringBuffer error("While reporting exception: ");
         EE->errorMessage(error);
-        DBGLOG("%s", error.str());
+        OERRLOG("%s", error.str());
```
Comment (ghalliday) [inline common/thorhelper/roxiehelper.cpp]: likewise not sure this is operator - probably programmer.  I suspect there should be quite a high threshold on something going to the operator - misconfigured system, fatal errors, kernel errors.

File: common/thorhelper/roxiehelper.cpp [hunk 4]
```diff
@@ -1696,7 +1696,7 @@ void CSafeSocket::sendJsonException(IException *E, const char *queryName)
     catch(IException *EE)
     {
         StringBuffer error("While reporting exception: ");
-        DBGLOG("%s", EE->errorMessage(error).str());
+        OERRLOG("%s", EE->errorMessage(error).str());
```
Comment (ghalliday) [inline common/thorhelper/roxiehelper.cpp]: same here, ERRLOG I think

File: common/thorhelper/thorsoapcall.cpp [hunk 5]
```diff
@@ -632,7 +632,7 @@ class CMatchCB : implements IXMLSelect, public CInterface
                 excptEntry->getStringX(len, message, "Message");
                 StringBuffer mstr(len, message);
                 rtlFree(message);
-                DBGLOG("Roxie exception: %s", mstr.str());
+                IERRLOG("Roxie exception: %s", mstr.str());
```
Comment (ghalliday) [inline common/thorhelper/thorsoapcall.cpp]: Why IERRLOG, when is that used?  Is it used for logging an error for the programmer.  Should it be DERRLOG?

File: common/workunit/workunit.cpp [hunk 6]
```diff
@@ -6917,7 +6917,7 @@ wuTokenStates verifyWorkunitDAToken(const char * distributedAccessToken)
         StringBuffer sig(++finger);
         if (!pDSM->digiVerify(sig, token))
         {
-            ERRLOG("verifyWorkunitDAToken : workunit distributed access token does not verify");
+            OERRLOG("verifyWorkunitDAToken : workunit distributed access token does not verify");
```
Comment (ghalliday) [inline common/workunit/workunit.cpp]: Not operator - can be triggerred by invalid request from a user.

File: common/workunit/workunit.cpp [hunk 7]
```diff
@@ -6934,7 +6934,7 @@ wuTokenStates verifyWorkunitDAToken(const char * distributedAccessToken)
     {
         if (!streq(cw->queryUser(), tokUser.str()))
         {
-            ERRLOG("verifyWorkunitDAToken : token user does not match workunit");
+            OERRLOG("verifyWorkunitDAToken : token user does not match workunit");
```
Comment (ghalliday) [inline common/workunit/workunit.cpp]: Not sure it is operator, possibly audit though - suggests something fishy going on.

File: common/workunit/workunit.cpp [hunk 8]
```diff
@@ -6955,7 +6955,7 @@ wuTokenStates verifyWorkunitDAToken(const char * distributedAccessToken)
         wuActive = true;
         break;
     default:
-        ERRLOG("verifyWorkunitDAToken : Workunit %s not active, state is '%s'", cw->queryWuid(), getWorkunitStateStr(cw->getState()));
+        OERRLOG("verifyWorkunitDAToken : Workunit %s not active, state is '%s'", cw->queryWuid(), getWorkunitStateStr(cw->getState()));
```
Comment (ghalliday) [inline common/workunit/workunit.cpp]: Probably user rather than operator.

**Comment by ghalliday** [comment]
Submitted comments so far since it got confused when you rebased.

File: common/thorhelper/thorsoapcall.cpp (hunk 5 above)
Comment (ghalliday) [reply common/thorhelper/thorsoapcall.cpp]: I'm fine with IERRLOG if we all know that means internal/programmer.

File: common/remote/sockfile.cpp (hunk 1 above)
Comment (jakesmith) [reply common/remote/sockfile.cpp]: @ghalliday - Agree. I don't think this should be an operator error. ERRLOG in a #ifDEF _DEBUG statement is fine.
As a general rule, I suspect anything inside a DEBUG section should not be an operator error.

File: system/security/LdapSecurity/ldapsecurity.cpp [hunk 9]
```diff
@@ -448,7 +448,7 @@ ISecResource* CLdapSecResourceList::addResource(const char * name)
 {
     if(!name || !*name)
     {
-        DBGLOG("CLdapSecResourceList::addResource resource name must be provided");
+        OWARNLOG("CLdapSecResourceList::addResource resource name must be provided");
```
Comment (ghalliday) [inline system/security/LdapSecurity/ldapsecurity.cpp]: These look like internal errors, not operator issues.

File: system/security/LdapSecurity/ldaputils.cpp [hunk 10]
```diff
@@ -76,7 +76,7 @@ LDAP* LdapUtils::LdapInit(const char* protocol, const char* host, int port, int
         {
             if (throwOnError)
                 throw MakeStringException(-1, "ldap_initialize error %s", ldap_err2string(rc));
-            DBGLOG("ldap_initialize error %s", ldap_err2string(rc));
```
Comment (ghalliday) [inline system/security/LdapSecurity/ldaputils.cpp]: russ can comment on which of these actually correspond to operator issues.

File: system/security/LdapSecurity/permissions.cpp [hunk 11]
```diff
@@ -182,12 +182,12 @@ void PermissionProcessor::getCachedSid(const char* name, MemoryBuffer& sid)
         {
             StringBuffer errmsg;
             e->errorMessage(errmsg);
-            DBGLOG("error getPropbin - %s", errmsg.str());
+            OERRLOG("error getPropbin - %s", errmsg.str());
```
Comment (ghalliday) [inline system/security/LdapSecurity/permissions.cpp]: I doubt any of these are operator errors - most likely internal

File: system/security/securesocket/securesocket.cpp [hunk 12]
```diff
@@ -840,7 +840,7 @@ int verify_callback(int ok, X509_STORE_CTX *store)
             ok = true;
         }
         else
-            DBGLOG("Error with certificate: issuer=%s,subject=%s,err %d - %s", issuer, subject,err,X509_verify_cert_error_string(err));
+            OERRLOG("Error with certificate: issuer=%s,subject=%s,err %d - %s", issuer, subject,err,X509_verify_cert_error_string(err));
```
Comment (ghalliday) [inline system/security/securesocket/securesocket.cpp]: I doubt this is an operator error - when is it reported?  Almost certainly internal - e.g., when connecting to some other system

File: system/security/shared/authmap.cpp [hunk 13]
```diff
@@ -21,7 +21,7 @@ int CAuthMap::add(const char* path, ISecResourceList* resourceList)
 {
     if(!path || !*path)
     {
-        DBGLOG("can't add to CAuthMap, path is NULL");
+        OWARNLOG("Can't add to CAuthMap, path is NULL");
```
Comment (ghalliday) [inline system/security/shared/authmap.cpp]: Why O?

File: system/xmllib/libxml_xpathprocessor.cpp [hunk 14]
```diff
@@ -241,15 +241,15 @@ class CLibXpathContext : public CInterface, public IXpathContext
             m_xmlDoc = xmlParseDoc((const unsigned char *)xmldoc);
             if (m_xmlDoc == nullptr)
             {
-                ERRLOG("XpathProcessor:setxmldoc Error: Unable to parse XMLLib document");
+                OERRLOG("XpathProcessor:setxmldoc Error: Unable to parse XMLLib document");
```
Comment (ghalliday) [inline system/xmllib/libxml_xpathprocessor.cpp]: Do you know the xml being processed is a system xmlfile?  No indication of what file, so almost certainly needs to be I

File: thorlcr/graph/thgraphmaster.cpp [hunk 15]
```diff
@@ -1156,12 +1156,12 @@ class CThorCodeContextMaster : public CThorCodeContextBase
         {
             StringBuffer m;
             E->errorMessage(m);
-            DBGLOG("Unable to record exception in workunit: %s", m.str());
+            OERRLOG("Unable to record exception in workunit: %s", m.str());
```
Comment (ghalliday) [inline thorlcr/graph/thgraphmaster.cpp]: debatable.  Jake?

File: tools/backupnode/backupnode.cpp [hunk 16]
```diff
@@ -62,7 +62,7 @@ static bool inexactDateMatch = false;
 
 #ifdef USE_JLOG // and why not?
 #define println PROGLOG
-#define printerr ERRLOG
+#define printerr OERRLOG
```
Comment (ghalliday) [inline tools/backupnode/backupnode.cpp]: Why this change? I think most (all?) relate to use specified options, so not relevant to notify the operator

File: tools/esdlcmd/esdlcmd_common.cpp [hunk 17]
```diff
@@ -65,7 +65,7 @@ void saveAsFile(const char * filepath, const char *filename, const char *text, c
     if (io.get())
         io->write(0, strlen(text), text);
     else
-        DBGLOG("File %s can't be created", file->queryFilename());
+        OERRLOG("File %s can't be created", file->queryFilename());
```
Comment (ghalliday) [inline tools/esdlcmd/esdlcmd_common.cpp]: A command line tool, so probably should be U

File: tools/esdlcmd/esdlcmd_shell.cpp [hunk 18]
```diff
@@ -126,7 +126,7 @@ int EsdlCMDShell::run()
 #ifndef _DEBUG
     catch (...)
     {
-        ERRLOG("Unexpected exception\n");
+        OERRLOG("Unexpected exception\n");
```
Comment (ghalliday) [inline tools/esdlcmd/esdlcmd_shell.cpp]: I is more appropriate I think

File: tools/swapnode/swapnodelib.cpp [hunk 19]
```diff
@@ -76,12 +76,12 @@ bool WuResubmit(const char *wuid)
     Owned<IWorkUnit> wu = factory->updateWorkUnit(wuid);
     if (!wu)
     {
-        ERRLOG("WuResubmit(%s): could not find workunit",wuid);
+        OERRLOG("WuResubmit(%s): could not find workunit",wuid);
```
Comment (ghalliday) [inline tools/swapnode/swapnodelib.cpp]: Isn't this most likely to be user error?  Sae for many of the rest

**Review by ghalliday — COMMENTED** [review]
@shamser I've had a first pass through the PR.  A lot is right, but I think the PR contains too many changes.  (Even GitHub is struggling slightly with the size.)

I think to get this merged we will need to split it up into multiple PRS.
The first PR can introduce the new functions.  Then have multiple PRs to switch warnings in different areas over to the new scheme.  (E.g., dali, thor, roxie, esp, jlib).  That way the people with knowledge of that particular area can check them.

A few general comments - if it is a test program then it should be programmer.  If it is from user input, especially command line then it should be user.

**Comment by AttilaVamos** [comment]
@AttilaVamos Those spray errors has been caused by my merged PR-https://github.com/hpcc-systems/HPCC-Platform/pull/12206 . I'm working on the fix. I am sorry about inconvenience caused.

**Comment by AttilaVamos** [comment]
@shamser back to you.
